### ISSUE TITLE
PyRevit 4.6.18 failed to launch in Revit 2017.2

### ISSUE THREAD

**Issue by Robbo1234** [eirannejad/pyRevit#543]
**Describe the bug**
Missing PyRevit 4.6.18 addin tab in Revit 2017.2

```
C:\Users\rob.cross>pyrevit env
==> Registered Clones (full git repos)
==> Registered Clones (deployed from archive)
master | Deploy: "basepublic" | Branch: "master" | Version: "4.6.18" | Path: "C:
\Users\rob.cross\AppData\Roaming\pyRevit-Master"
==> Attachments
Error: Version string portion was too short or too long.
Run with "--debug" option to see debug messages

C:\Users\rob.cross>pyrevit attach master 277 --installed
Error: Version string portion was too short or too long.
Run with "--debug" option to see debug messages

C:\Users\rob.cross>pyrevit attach master 277 --installed
Error: Version string portion was too short or too long.
Run with "--debug" option to see debug messages
```

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows 7 
 - pyRevit Version 4.6.18
 - pyRevit Environment: Open a command prompt and run the command below. Replace these lines with the results. Take a look at this markdown guide and wrap the command results in ``` when pasting here for correct formatting.

```
C:\Users\rob.cross>pyrevit env
==> Registered Clones (full git repos)
==> Registered Clones (deployed from archive)
master | Deploy: "basepublic" | Branch: "master" | Version: "4.6.18" | Path: "C:
\Users\rob.cross\AppData\Roaming\pyRevit-Master"
==> Attachments
Error: Version string portion was too short or too long.
Run with "--debug" option to see debug messages
```

**Comment by Robbo1234** [comment]
When I was uninstalled pyRevit 4.6.18 and reinstalled pyRevit 4.6.17 I noticed that PyRevit gave me a "select location" dialog option but  it didn't give me that option when installing pyRevit Version 4.6.18. Refer to screen shot.

Note: pyRevit Version 4.6.17 opened up fine again in Revit 2017.2 on my office PC plus pyRevit 4.6.18 loaded up fine on my laptop.

![image](https://user-images.githubusercontent.com/8776947/54378145-8ef8bb80-467e-11e9-8dff-f5d9cbac7a7a.png)

**Comment by Robbo1234** [comment]
Hi Ehsan,

Is there a solution for this?

Thanking you in advance

Rob

**Comment by eirannejad** [comment]
@Robbo1234 Run `pyrevit env --debug` and share all the results please. Thanks

**Comment by Robbo1234** [comment]
Debug: Analysing registered app: Microsoft Visual C++ 2010  x64 Redistributable
- 10.0.40219
Debug: Analysing registered app: Microsoft Visual C++ 2013 x64 Additional Runtim
e - 12.0.40649
Debug: Analysing registered app: pyRevit
Debug: Analysing registered app: Intel® Trusted Connect Service Client
Debug: Analysing registered app: Microsoft Visual C++ 2017 x64 Minimum Runtime -
 14.13.26020
Debug: Analysing registered app: SQL Server 2008 R2 SP2 Common Files
Debug: Analysing registered app: Autodesk Workflows 2017
Debug: Analysing registered app: Java 8 Update 121 (64-bit)
Debug: Analysing registered app: Autodesk BIM 360 Glue AutoCAD 2017 Add-in 64 bi
t
Debug: Analysing registered app:
Debug: Analysing registered app: Microsoft SQL Server VSS Writer
Debug: Analysing registered app: Autodesk Workflows 2018
Debug: Analysing registered app: AutoCAD 2017
Debug: Analysing registered app: ACAD Private
Debug: Analysing registered app: AutoCAD 2017 Language Pack - English
Debug: Analysing registered app: AutoCAD 2017 - English
Debug: Analysing registered app: ACA & MEP 2017 Object Enabler
Debug: Analysing registered app: DWG TrueView 2017 - English
Debug: Analysing registered app: DWG TrueView 2019 - English
Debug: Analysing registered app: Microsoft .NET Core Host FX Resolver - 2.0.7 (x
64)
Debug: Analysing registered app: Dynamo Core 1.0.0
Debug: Analysing registered app: Autodesk License Service (x64) - 5.1.4
Debug: Analysing registered app: SQL Server 2008 R2 SP2 Common Files
Debug: Analysing registered app: Microsoft Visual C++ 2012 x64 Additional Runtim
e - 11.0.61030
Debug: Analysing registered app: Autodesk BIM 360 Revit 2015 Add-in 64 bit
Debug: Analysing registered app: Autodesk Navisworks Manage 2016 - 2014 DWG File
 Reader
Debug: Analysing registered app: Autodesk Navisworks 2016 Exporters - 64 bit - E
nglish Language Pack
Debug: Analysing registered app: Autodesk Navisworks 2016 Exporters - 64 bit
Debug: Analysing registered app: ESET Endpoint Antivirus
Debug: Analysing registered app: ESET Management Agent
Debug: Analysing registered app: Autodesk SketchBook Designer 2014
Debug: Analysing registered app: Microsoft SQL Server 2012 Management Objects  (
x64)
Debug: Analysing registered app: eTransmit for Autodesk Revit 2015
Debug: Analysing registered app: eTransmit for Autodesk Revit 2016
Debug: Analysing registered app: eTransmit for Autodesk Revit 2017
Debug: Analysing registered app: eTransmit for Autodesk Revit 2018
Debug: Analysing registered app: eTransmit for Autodesk Revit 2019
Debug: Analysing registered app: Microsoft .NET Core Host - 2.0.7 (x64)
Debug: Analysing registered app: Autodesk Navisworks Manage 2016 - 2016 DWG File
 Reader
Debug: Analysing registered app: Autodesk Navisworks Manage 2017
Debug: Analysing registered app: Autodesk Navisworks Manage 2017 - English Langu
age Pack
Debug: Analysing registered app: FormIt
Debug: Analysing registered app: Microsoft Visual C++ 2008 Redistributable - x64
 9.0.30729.4148
Debug: Analysing registered app: Intel(R) ME UninstallLegacy
Debug: Analysing registered app: Autodesk Navisworks 2019 Exporters - 64 bit - E
nglish Language Pack
Debug: Analysing registered app: Autodesk Navisworks 2019 Exporters - 64 bit
Debug: Analysing registered app: Intel(R) Management Engine Components
Debug: Analysing registered app: Worksharing Monitor for Autodesk Revit 2016
Debug: Analysing registered app: Worksharing Monitor for Autodesk Revit 2017
Debug: Analysing registered app: Worksharing Monitor for Autodesk Revit 2018
Debug: Analysing registered app: Worksharing Monitor for Autodesk Revit 2019
Debug: Analysing registered app: Autodesk ReCap
Debug: Analysing registered app: SQL Server 2008 R2 SP2 Management Studio
Debug: Analysing registered app: Microsoft SQL Server 2014 Express LocalDB
Debug: Analysing registered app: Autodesk Workflows 2016
Debug: Analysing registered app: Autodesk Navisworks Manage 2018
Debug: Analysing registered app: Autodesk Navisworks Manage 2018 - English Langu
age Pack
Debug: Analysing registered app: AutoCAD 2014 - English
Debug: Analysing registered app: AutoCAD 2014 Language Pack - English
Debug: Analysing registered app: AutoCAD 2014 - English
Debug: Analysing registered app: Autodesk InfraWorks 360
Debug: Analysing registered app: Autodesk InfraWorks 360 Language Pack
Debug: Analysing registered app: Autodesk InfraWorks 360 Content
Debug: Analysing registered app: Microsoft .NET Core Runtime - 2.0.7 (x64)
Debug: Analysing registered app: Windows Small Business Server 2011 Standard Cli
entAgent
Debug: Analysing registered app: Autodesk Navisworks 2018 Exporters - 64 bit - E
nglish Language Pack
Debug: Analysing registered app: Autodesk Navisworks 2018 Exporters - 64 bit
Debug: Analysing registered app: FormIt Converter For Revit 2019
Debug: Analysing registered app: Microsoft Visual C++ 2008 Redistributable - x64
 9.0.30729.6161
Debug: Analysing registered app: Dynamo Revit 1.3.4
Debug: Analysing registered app: FormIt 360 Converter For Revit 2017
Debug: Analysing registered app: RTV Shared Parameter Manager 2016
Debug: Analysing registered app: Backup and Sync from Google
Debug: Analysing registered app: Microsoft Visual C++ 2005 Redistributable (x64)

Debug: Analysing registered app: LibreOffice 6.1.3.2
Debug: Analysing registered app: Autodesk Revit Model Review 2016
Debug: Analysing registered app: Autodesk Revit Model Review 2017
Debug: Analysing registered app: Autodesk Revit Model Review 2018
Debug: Analysing registered app: Autodesk Revit Model Review 2019
Debug: Analysing registered app: SQL Server 2008 R2 SP2 Management Studio
Debug: Analysing registered app: Revit 2014
Debug: App is a Revit product: Revit 2014
Debug: Read version from registery key: "13.11.00004"
Debug: Read install path from registery key: "C:\Program Files\Autodesk\Revit 20
14"
Debug: Read language code from registery key: "1033"
Debug: Could not determine Revit Product from version "13.11.00004"
Debug: Read build number "20140709_2115(x64)" from binary at "C:\Program Files\A
utodesk\Revit 2014\Revit.exe"
Debug: Revit Product is : {0}
Debug: Analysing registered app: Revit 2014 Language Pack - English
Debug: Analysing registered app: Revit 2015
Debug: App is a Revit product: Revit 2015
Debug: Read version from registery key: "15.0.1259.0"
Debug: Read install path from registery key: "C:\Program Files\Autodesk\Revit 20
15"
Debug: Read language code from registery key: "1033"
Debug: Could not determine Revit Product from version "15.0.1259.0"
Debug: Read build number "20160512_0715(x64)" from binary at "C:\Program Files\A
utodesk\Revit 2015\Revit.exe"
Debug: Expected path C:\Program Files\Autodesk\Revit 2015
Debug: Revit Product is : Autodesk Revit 2015 Release 2 Update Release 14  (Subs
cription only release) | Version: 15.2.14 | Build: 20160512_0715(x64) | Language
: 0 | Path: "C:\Program Files\Autodesk\Revit 2015"
Debug: Analysing registered app: Revit 2015 Language Pack - English
Debug: Analysing registered app: Revit 2016
Debug: App is a Revit product: Revit 2016
Debug: Read version from registery key: "16.0.1185.0"
Debug: Read install path from registery key: "C:\Program Files\Autodesk\Revit 20
16"
Debug: Read language code from registery key: "1033"
Debug: Expected path C:\Program Files\Autodesk\Revit 2016
Debug: Revit Product is : Autodesk Revit 2016 Update 7 for R2 | Version: 16.0.11
85.0 | Build: 20161004_0715(x64) | Language: 0 | Path: "C:\Program Files\Autodes
k\Revit 2016"
Debug: Analysing registered app: Revit 2017
Debug: App is a Revit product: Revit 2017
Debug: Read version from registery key: "17.0.1150.0"
Debug: Read install path from registery key: "C:\Program Files\Autodesk\Revit 20
17"
Debug: Read language code from registery key: "1033"
Debug: Expected path C:\Program Files\Autodesk\Revit 2017
Debug: Revit Product is : Autodesk Revit 2017.2.3 | Version: 17.0.1150.0 | Build
: 20171027_0315(x64) | Language: 0 | Path: "C:\Program Files\Autodesk\Revit 2017
"
Debug: Analysing registered app: Revit 2018
Debug: App is a Revit product: Revit 2018
Debug: Read version from registery key: "18.3.1.2"
Debug: Read install path from registery key: "C:\Program Files\Autodesk\Revit 20
18"
Debug: Read language code from registery key: "1033"
Debug: Expected path C:\Program Files\Autodesk\Revit 2018
Debug: Revit Product is : Autodesk Revit 2018.3.1 | Version: 18.3.1.2 | Build: 2
0180423_1000(x64) | Language: 0 | Path: "C:\Program Files\Autodesk\Revit 2018"
Debug: Analysing registered app: Revit 2019
Debug: App is a Revit product: Revit 2019
Debug: Read version from registery key: "19.2.0.65"
Debug: Read install path from registery key: "C:\Program Files\Autodesk\Revit 20
19"
Debug: Read language code from registery key: "1033"
Debug: Expected path C:\Program Files\Autodesk\Revit 2019
Debug: Revit Product is : Autodesk Revit 2019.2 (Update) | Version: 19.2.0.65 |
Build: 20181217_1515(x64) | Language: 0 | Path: "C:\Program Files\Autodesk\Revit
 2019"
Debug: Analysing registered app: BricsCAD Shape V18.2.02 (x64) en_US
Debug: Analysing registered app: Intel(R) Management Engine Components
Debug: Analysing registered app: Microsoft SQL Server 2008 R2 Native Client
Debug: Analysing registered app: Autodesk Certificate Package  (x64) - 5.1.4
Debug: Analysing registered app: Compare Models for Revit 2014
Debug: Analysing registered app: Compare Models for Revit
Debug: Analysing registered app: Autodesk Revit 2019 MEP Fabrication Configurati
on - Imperial
Debug: Analysing registered app:
Debug: Analysing registered app:
Debug: Analysing registered app:
Debug: Analysing registered app: Personal Accelerator for Revit
Debug: Analysing registered app: Inkscape 0.92.3
Debug: Analysing registered app:
Debug: Analysing registered app: Batch Print for Autodesk Revit 2018
Debug: Analysing registered app: Batch Print for Autodesk Revit 2019
Debug: Analysing registered app: Autodesk Navisworks Manage 2016 - 2010 DWG File
 Reader
Debug: Analysing registered app: Autodesk Navisworks Manage 2019 - DWG File Read
er
Debug: Analysing registered app: Intel(R) Chipset Device Software
Debug: Analysing registered app: Revit Extensions for Autodesk Revit 2017
Debug: Analysing registered app: Revit Extensions for Autodesk Revit 2016
Debug: Analysing registered app: Autodesk Revit 2019 MEP Fabrication Configurati
on - Metric
Debug: Analysing registered app: Office 16 Click-to-Run Licensing Component
Debug: Analysing registered app: Office 16 Click-to-Run Extensibility Component
64-bit Registration
Debug: Analysing registered app: Microsoft .NET Framework 4.7.2
Debug: Analysing registered app: Revit Content Libraries 2018
Debug: Analysing registered app: Revit Content Libraries 2019
Debug: Analysing registered app: Microsoft Application Error Reporting
Debug: Analysing registered app: Dynamo Studio 2016
Debug: Analysing registered app: Microsoft System CLR Types for SQL Server 2012
(x64)
Debug: Analysing registered app: Autodesk Navisworks Manage 2016
Debug: Analysing registered app: Autodesk Navisworks Manage 2016 - English Langu
age Pack
Debug: Analysing registered app: BIMCAVEAutoSearchSet
Debug: Analysing registered app: FormIt Converter For Revit 2018
Debug: Analysing registered app: SQL Server 2008 R2 SP2 Database Engine Shared
Debug: Analysing registered app: Grids Generator for Revit 2014
Debug: Analysing registered app: Grids Generator for Revit
Debug: Analysing registered app: Dynamo Revit 2.0.2
Debug: Analysing registered app: pyRevit CLI
Debug: Analysing registered app: Autodesk Workflows 2015
Debug: Analysing registered app:
Debug: Analysing registered app: Autodesk A360 Collaboration v8 for Revit 2016
Debug: Analysing registered app:
Debug: Analysing registered app:
Debug: Analysing registered app:
Debug: Analysing registered app: Autodesk A360 Collaboration for Revit 2017
Debug: Analysing registered app:
Debug: Analysing registered app:
Debug: Analysing registered app: Autodesk Collaboration for Revit 2018
Debug: Analysing registered app: Autodesk Collaboration for Revit 2019
Debug: Analysing registered app: Microsoft Visual C++ 2013 x64 Minimum Runtime -
 12.0.40649
Debug: Analysing registered app: Microsoft Visual C++ 2005 Redistributable (x64)

Debug: Analysing registered app: Dynamo Core 2.0.2
Debug: Analysing registered app: Autodesk Navisworks Manage 2016 - 2011 DWG File
 Reader
Debug: Analysing registered app: NVIDIA Ansel
Debug: Analysing registered app: NVIDIA 3D Vision Driver 392.00
Debug: Analysing registered app: NVIDIA Control Panel 392.00
Debug: Analysing registered app: NVIDIA Graphics Driver 392.00
Debug: Analysing registered app: NVIDIA nView 149.34
Debug: Analysing registered app: NVIDIA 3D Vision Controller Driver 390.41
Debug: Analysing registered app: NVIDIA WMI 2.31.0
Debug: Analysing registered app: NVIDIA HD Audio Driver 1.3.36.6
Debug: Analysing registered app: NVIDIA Install Application
Debug: Analysing registered app: NVIDIA Display Container
Debug: Analysing registered app: NVIDIA Display Container LS
Debug: Analysing registered app: NVIDIA Display Watchdog Plugin
Debug: Analysing registered app: NVIDIA Display Session Container
Debug: Analysing registered app: Microsoft SQL Server 2008 Setup Support Files

Debug: Analysing registered app: Autodesk Navisworks Manage 2018 - 2018 DWG File
 Reader
Debug: Analysing registered app: Autodesk Dynamo Studio 2017
Debug: Analysing registered app: Intel(R) Management Engine Components
Debug: Analysing registered app: Autodesk Navisworks Manage 2019
Debug: Analysing registered app: Autodesk Navisworks Manage 2019 - English Langu
age Pack
Debug: Analysing registered app: Microsoft SQL Server 2008 R2 Setup (English)
Debug: Analysing registered app: Autodesk Navisworks Manage 2017 - 2017 DWG File
 Reader
Debug: Analysing registered app: Autodesk Navisworks Manage 2016 - 2012 DWG File
 Reader
Debug: Analysing registered app: Microsoft Visual C++ 2017 x64 Additional Runtim
e - 14.13.26020
Debug: Analysing registered app: SQL Server 2008 R2 SP2 Database Engine Shared
Debug: Analysing registered app: Microsoft Visual C++ 2012 x64 Minimum Runtime -
 11.0.61030
Debug: Analysing registered app: TightVNC
Debug: Analysing registered app: Microsoft SQL Server 2008 R2 RsFx Driver
Debug: Analysing registered app: Dynamo Core 1.3.4
Debug: Analysing registered app: Microsoft Lync Web App Plug-in
Debug: Analysing registered app: Autodesk Navisworks Manage 2016 - 2013 DWG File
 Reader
Debug: Analysing registered app: SketchUp 2017
Debug: Analysing registered app: Bonus Tools Licence Manager
Debug: Analysing registered app: Autodesk License Service (x64) - 3.1
Debug: Analysing registered app: RTV Xporter 2016
Debug: Analysing registered app: Revit Extensions for Autodesk Revit 2018
Debug: Analysing registered app: WebDrive
Debug: Analysing registered app: Dynamo Core 1.3.3
Debug: Analysing registered app: Sql Server Customer Experience Improvement Prog
ram
Debug: Analysing registered app: Autodesk License Service (x64) - 7.1.4
Debug: Analysing registered app: PowerPack for Revit
Debug: Analysing registered app: Google Earth Pro
Debug: Analysing registered app: Autodesk Workflows 2019
Debug: Analysing registered app: UE4 Prerequisites (x64)
Debug: Analysing registered app: Revit Extensions for Autodesk Revit 2014
Debug: Analysing registered app: SQL Server 2008 R2 SP2 Database Engine Services

Debug: Analysing registered app: SQL Server 2008 R2 SP2 Database Engine Services

Debug: Analysing registered app: novaPDF 8 add-in for Microsoft Office (x64)
Debug: Analysing registered app: Microsoft Power BI Desktop (x64)
Debug: Analysing registered app: PowerPack for Revit
Error: Version string portion was too short or too long. (System.ArgumentExcepti
on)
   at System.Version.VersionResult.SetFailure(ParseFailureKind failure, String a
rgument)
   at System.Version.TryParseVersion(String version, VersionResult& result)
   at System.Version.Parse(String input)
   at System.Version..ctor(String version)
   at pyRevitLabs.TargetApps.Revit.RevitProduct.get_Version()
   at pyRevitLabs.TargetApps.Revit.PyRevit.GetAttachments()
   at pyRevitManager.Views.pyRevitCLI.PrintAttachments()
   at pyRevitManager.Views.pyRevitCLI.ExecuteCommand(IDictionary`2 arguments, IE
numerable`1 activeKeys)
   at pyRevitManager.Views.pyRevitCLI.ProcessArguments(String[] args)

C:\Users\rob.cross>

**Comment by eirannejad** [comment]
Okay. A few missing Revit version and build numbers. Will fix for 4.6.18. Meanwhile copy the `pyRevit.addin` manifest file manually into `%appdata%/Autodesk/Revit/Addons/2017` folder so pyRevit loads.

**Comment by eirannejad** [comment]
Fixed.
